samples/drivers/i2s/i2s_codec/src/main.c

```diff
@@ -11,21 +11,24 @@
 #include <zephyr/audio/codec.h>
 #include <string.h>
 
+#ifndef CONFIG_USE_DMIC
+#include "sine.h"
+#endif
 
-#define I2S_CODEC_TX  DT_ALIAS(i2s_codec_tx)
+#define I2S_CODEC_TX DT_ALIAS(i2s_codec_tx)
 
-#define SAMPLE_FREQUENCY    CONFIG_SAMPLE_FREQ
-#define SAMPLE_BIT_WIDTH    (16U)
-#define BYTES_PER_SAMPLE    sizeof(int16_t)
+#define SAMPLE_FREQUENCY CONFIG_SAMPLE_FREQ
+#define SAMPLE_BIT_WIDTH (16U)
+#define BYTES_PER_SAMPLE sizeof(int16_t)
 #if CONFIG_USE_DMIC
-#define NUMBER_OF_CHANNELS  CONFIG_DMIC_CHANNELS
+#define NUMBER_OF_CHANNELS CONFIG_DMIC_CHANNELS
 #else
```

Comment:
Please do not add style changes in this patch. This makes the real change harder to review. If you really want to do style changes please create a separate patch.

Also, split this patch further.

- one to add sine wave support
- one to add support mimxrt1170 evkb and wm8962 codec